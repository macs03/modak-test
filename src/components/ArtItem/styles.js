import { StyleSheet } from "react-native";
import { Colors, BorderRadiuses } from "react-native-ui-lib";

const styles = StyleSheet.create({
  image: {
    width: 54,
    height: 54,
    borderRadius: BorderRadiuses.br20,
    marginHorizontal: 14
  },
  border: {
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: Colors.grey70
  }
});

export default styles;
