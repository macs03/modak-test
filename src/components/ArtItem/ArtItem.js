import { Colors, View, Image, ListItem, Text } from "react-native-ui-lib";
import { Alert } from "react-native";

import styles from "./styles";

function ArtItem({ item, onPressItem }) {
  return (
    <View>
      <ListItem
        activeBackgroundColor={Colors.grey60}
        activeOpacity={0.3}
        height={77.5}
        onPress={() => onPressItem(item.id)}
      >
        <ListItem.Part left>
          <Image
            source={{
              uri: `https://www.artic.edu/iiif/2/${item.image_id}/full/200,/0/default.jpg`
            }}
            style={styles.image}
          />
        </ListItem.Part>
        <ListItem.Part
          middle
          column
          containerStyle={[styles.border, { paddingRight: 17 }]}
        >
          <ListItem.Part containerStyle={{ marginBottom: 3 }}>
            <Text
              grey10
              text70
              style={{ flex: 1, marginRight: 10 }}
              numberOfLines={1}
            >
              {item.artist_display}
            </Text>
            <Text grey10 text70 style={{ marginTop: 2 }}>
              {item.date_display}
            </Text>
          </ListItem.Part>
          <ListItem.Part>
            <Text text90 numberOfLines={1}>
              {item?.thumbnail?.alt_text}
            </Text>
          </ListItem.Part>
        </ListItem.Part>
      </ListItem>
    </View>
  );
}

export default ArtItem;
