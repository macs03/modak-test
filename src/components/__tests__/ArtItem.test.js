import * as React from "react";
import renderer from "react-test-renderer";

import ArtItem from "../ArtItem/ArtItem";

it(`renders correctly`, () => {
  const item = {
    image_id: 345,
    artist_display: "tests",
    date_display: "tests",
    thumbnail: {
      alt_text: "test"
    }
  };

  const tree = renderer
    .create(<ArtItem item={item} onPressItem={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
