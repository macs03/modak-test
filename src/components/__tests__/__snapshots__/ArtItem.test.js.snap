// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "flexDirection": "row",
              "height": 77.5,
            },
            undefined,
            false,
          ],
        ]
      }
    >
      <View
        modifiers={
          Object {
            "alignments": Object {
              "alignItems": "flex-start",
            },
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "color": undefined,
            "flexStyle": undefined,
            "margins": Object {},
            "paddings": Object {},
            "positionStyle": undefined,
            "typography": undefined,
          }
        }
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {
              "alignItems": "flex-start",
            },
            Array [
              Object {
                "alignItems": "center",
                "flex": 0,
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              undefined,
            ],
          ]
        }
      >
        <Image
          accessibilityRole="image"
          accessible={false}
          assetGroup="icons"
          onError={[Function]}
          source={
            Object {
              "uri": "https://www.artic.edu/iiif/2/345/full/200,/0/default.jpg",
            }
          }
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              false,
              Object {
                "borderRadius": 6,
                "height": 54,
                "marginHorizontal": 14,
                "width": 54,
              },
              false,
            ]
          }
        />
      </View>
      <View
        column={true}
        middle={true}
        modifiers={
          Object {
            "alignments": Object {},
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "color": undefined,
            "flexStyle": undefined,
            "margins": Object {},
            "paddings": Object {},
            "positionStyle": undefined,
            "typography": undefined,
          }
        }
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {},
            Array [
              Object {
                "alignItems": undefined,
                "flex": 1,
                "flexDirection": undefined,
                "justifyContent": "center",
              },
              Array [
                Object {
                  "borderBottomWidth": 0.5,
                  "borderColor": "#F0F2F5",
                },
                Object {
                  "paddingRight": 17,
                },
              ],
            ],
          ]
        }
      >
        <View
          modifiers={Object {}}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                Object {
                  "marginBottom": 3,
                },
              ],
            ]
          }
        >
          <Text
            grey10={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#20303C",
                },
                undefined,
                undefined,
                Object {},
                undefined,
                undefined,
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 10,
                },
              ]
            }
            text70={true}
          >
            tests
          </Text>
          <Text
            grey10={true}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#20303C",
                },
                undefined,
                undefined,
                Object {},
                undefined,
                undefined,
                undefined,
                Object {
                  "marginTop": 2,
                },
              ]
            }
            text70={true}
          >
            tests
          </Text>
        </View>
        <View
          modifiers={Object {}}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "700",
                  "lineHeight": 16,
                },
                undefined,
                undefined,
                undefined,
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
            text90={true}
          >
            test
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
